transformer_engine/pytorch/csrc/extensions/dropout.cpp

```diff
@@ -0,0 +1,133 @@
+/*************************************************************************
+ * Copyright (c) 2022-2025, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ *
+ * See LICENSE for license information.
+ ************************************************************************/
+
+#include "transformer_engine/dropout.h"
+
+#include <ATen/cuda/CUDAGeneratorImpl.h>
+
+#include <ATen/cuda/CUDAGraphsUtils.cuh>
+
+#include "common.h"
+#include "pybind.h"
+#include "transformer_engine/transformer_engine.h"
+
+namespace transformer_engine::pytorch {
+
+void unpack(at::PhiloxCudaState arg, int64_t *rng_state_ptr) {
+  NVTE_SCOPED_GIL_RELEASE({
+    nvte_extract_seed_and_offset(rng_state_ptr, arg.captured_, arg.seed_.ptr, arg.seed_.val,
+                                 arg.offset_.ptr, arg.offset_.val, arg.offset_intragraph_,
+                                 at::cuda::getCurrentCUDAStream());
+  });
+}
+
+std::vector<py::object> dropout_fwd(const at::Tensor &input, float dropout_probability,
+                                    bool is_training) {
+  using namespace transformer_engine::pytorch::detail;
+
+  auto input_tensor = makeTransformerEngineTensor(input);
+  const int rng_block_size = 16;
+  int numel = input.numel();
+
+  // Allocate output tensor
+  std::vector<int> output_shape;
+  for (auto s : input.sizes()) {
+    output_shape.emplace_back(static_cast<int>(s));
+  }
+  auto output = allocateTorchTensor(output_shape[0], output_shape[1], input_tensor.dtype());
+  //auto output = at::empty(output_shape, input_tensor.dtype());
+  auto output_tensor = makeTransformerEngineTensor(output);
+
+  // Allocate mask tensor
+  std::vector<size_t> mask_shape{input.numel() / 16};
+  auto mask = allocateTorchTensor(mask_shape[0], DType::kInt16);
+  //auto mask = at::empty(mask_shape, at::CUDA(at::kUInt16));
```

Comment:
Why are there some commented out lines of code?

Reply:
Fixed.